tests/fs_remount.rs

```diff
@@ -0,0 +1,29 @@
+use std::fs;
+
+use birdcage::{Birdcage, Exception, Sandbox};
+
+fn main() {
+    const FILE_CONTENT: &str = "expected content";
+
+    // Setup our test tree.
+    let tempdir = tempfile::tempdir().unwrap().into_path();
+    let tempfile = tempdir.join("target-file");
+    fs::write(&tempfile, FILE_CONTENT.as_bytes()).unwrap();
+
+    // Setup sandbox, allowing write to dir, but only read for the file.
+    let mut birdcage = Birdcage::new();
+    birdcage.add_exception(Exception::WriteAndRead(tempdir.clone())).unwrap();
+    birdcage.add_exception(Exception::Read(tempfile.clone())).unwrap();
```

Comment:
Having a read-only path within a read-write path doesn't work on macOS (which is why this test is failing there). I didn't think it worked on Linux either when we were using Landlock, but I can't recall specifically.

I also thought this was documented, but the closest I can find is [the doc comment on `add_exception()`][1]...

> Once an exception is added, it is **not** possible to prohibit access to this resource without creating a new sandbox.

This can be a separate issue, but we'll probably need to either:

1. Change the behavior on Linux to make Birdcage work consistently across platforms
2. Document the inconsistent behavior
3. Force this to be an error or something? Not sure if that's practical though, especially with symlinks...

[1]: https://github.com/phylum-dev/birdcage/blob/c2655d833a59fcae975f4ed182c66f3a69fdc3a0/src/lib.rs#L63-L65